### PR TITLE
Convert tree-based Fed XGBoost with Job API

### PR DESCRIPTION
**Pull request by chesterxgchen**
Fixes # .

### Description

A few sentences describing the changes proposed in this pull request.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

**Review by YuanTingHsieh — APPROVED** [review]
LGTM

File: examples/hello-world/step-by-step/higgs/xgboost/xgboost_htree_job.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+from nvflare.app_common.workflows.scatter_and_gather import ScatterAndGather
+from nvflare.app_opt.xgboost.tree_based.bagging_aggregator import XGBBaggingAggregator
+from nvflare.app_opt.xgboost.tree_based.model_persistor import XGBModelPersistor
+from nvflare.app_opt.xgboost.tree_based.shareable_generator import XGBModelShareableGenerator
+from nvflare.job_config.script_runner import FrameworkType, ScriptRunner
+
+if __name__ == "__main__":
+    n_clients = 3
+    num_rounds = 100
+    train_script = "code/xgboost_fl.py"
```
Comment (YuanTingHsieh) [inline examples/hello-world/step-by-step/higgs/xgboost/xgboost_htree_job.py:24]: not sure if this still work for newest changes, please test it again

File: examples/hello-world/step-by-step/higgs/xgboost/xgboost_htree_job.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        train_task_name="train",
+        train_timeout=0,
+        allow_empty_global_weights=True,
+    )
+
+    job.to(ctrl, "server")
+
+    # Add clients
+    for i in range(n_clients):
+        executor = ScriptRunner(script=train_script, script_args=script_args, framework=FrameworkType.RAW)
```
Comment (SYangster) [inline examples/hello-world/step-by-step/higgs/xgboost/xgboost_htree_job.py:53]: suggest rename to runner?

**Review by SYangster — APPROVED** [review]

**Pull request merged**